tools/src/test/java/org/apache/kafka/tools/other/ReplicationQuotasTestRig.java

```diff
@@ -0,0 +1,479 @@
+        );
+        experiments.forEach(def -> run(def, journal, displayChartsOnScreen));
+
+        if (!displayChartsOnScreen)
+            Exit.exit(0, Option.empty());
+    }
+
+    static void run(ExperimentDef config, Journal journal, boolean displayChartsOnScreen) {
+        Experiment experiment = new Experiment();
+        try {
+            experiment.setUp(new EmptyTestInfo());
+            experiment.run(config, journal, displayChartsOnScreen);
+            journal.footer();
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            experiment.tearDown();
+        }
+    }
+
+    static class ExperimentDef {
+        String name;
+        int brokers;
+        int partitions;
+        long throttle;
+        int msgsPerPartition;
+        int msgSize;
+
+        public ExperimentDef(String name, int brokers, int partitions, long throttle, int msgsPerPartition, int msgSize) {
+            this.name = name;
+            this.brokers = brokers;
+            this.partitions = partitions;
+            this.throttle = throttle;
+            this.msgsPerPartition = msgsPerPartition;
+            this.msgSize = msgSize;
+        }
+
+        long targetBytesPerBrokerMB() {
+            return (long) msgsPerPartition * (long) msgSize * (long) partitions / brokers / 1_000_000;
+        }
+    }
+
+    static class Experiment extends QuorumTestHarness {
+        static final String TOPIC_NAME = "my-topic";
+
+        String experimentName = "unset";
+        List<KafkaServer> servers;
+        Map<Integer, List<Double>> leaderRates = new HashMap<>();
+        Map<Integer, List<Double>> followerRates = new HashMap<>();
+        Admin adminClient;
+
+        void startBrokers(List<Integer> brokerIds) {
+            System.out.println("Starting Brokers");
+            servers = brokerIds.stream().map(i -> createBrokerConfig(i, zkConnect()))
+                .map(c -> TestUtils.createServer(KafkaConfig.fromProps(c), Time.SYSTEM))
+                .collect(Collectors.toList());
+
+            TestUtils.waitUntilBrokerMetadataIsPropagated(seq(servers), DEFAULT_MAX_WAIT_MS);
+            String brokerList = TestUtils.plaintextBootstrapServers(seq(servers));
+            adminClient = Admin.create(Collections.singletonMap(
+                AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList
+            ));
+        }
+
+        @Override public void tearDown() {
+            Utils.closeQuietly(adminClient, "adminClient");
+            TestUtils.shutdownServers(seq(servers), true);
+            super.tearDown();
+        }
+
+        @SuppressWarnings({"unchecked", "deprecation"})
+        public void run(ExperimentDef config, Journal journal, boolean displayChartsOnScreen) throws Exception {
+            experimentName = config.name;
+            List<Integer> brokers = IntStream.rangeClosed(100, 100 + config.brokers).boxed().collect(Collectors.toList());
+            int[] count = new int[] {0};
```

Comment:
why is this an array?

Reply:
because only effectively final variable can be used inside java lambda - `IntSupplier nextReplicaRoundRobin`.
Reworked count as an instance variable of `nextReplicaRoundRobin`